### PR TITLE
Gracefully handle expr_text(<symbol>).

### LINKED ISSUE #611: quo_text() for bare symbols
``` r
library(rlang)
s <- withr::with_locale( c(LC_CTYPE = "C"), sym("\u4e2d") )
.Internal(inspect(quo_text(s)))
#> @7fcea4ae9c08 16 STRSXP g0c1 [] (len=1, tl=0)
#>   @7fcea4c706c8 09 CHARSXP g0c2 [gp=0x61] [ASCII] [cached] "<U+4E2D>"
.Internal(inspect(as_string(s)))
#> @7fcea4b138c0 16 STRSXP g0c1 [NAM(3)] (len=1, tl=0)
#>   @7fcea4d206c8 09 CHARSXP g0c1 [MARK,gp=0x28,ATT] [UTF8] [cached] "中"
```

This is I believe the issue in https://github.com/tidyverse/dplyr/issues/3829

I have a local fix in dplyr: https://github.com/tidyverse/dplyr/commit/b8a96babcc6ba81b3ded363d98b40d18bc1d42ee

### PR DESCRIPTION
**Pull request by romainfrancois**
When the expression is a bare symbol, rather use as_string and its magic handling of encoding. closes #611

Same fix as https://github.com/tidyverse/dplyr/pull/3830 but potentially benefitting outside of dplyr.

### CODE REVIEW

**Comment by lionel-** [comment]
Can you add a test for ``expr_text(quote(`<U+006F>`))`` please?

File: tests/testthat/test-quo.R [hunk 1]
```diff
@@ -128,6 +128,15 @@ test_that("quosures are spliced", {
   expect_identical(quo_text(q), "foo(bar(baz))")
 })
 
+test_that("quo_text() uses as_string encoding repair (#611)", {
+  old <- Sys.getlocale("LC_CTYPE")
+  on.exit(Sys.setlocale("LC_CTYPE", old))
```
Comment (lionel-) [inline tests/testthat/test-quo.R]: Probably best to avoid locale in the test, just create a symbol of a unicode tag.

Comment (romainfrancois) [reply tests/testthat/test-quo.R]: Right. I’ll clean it after 🍝.

**Comment by lionel-** [comment]
Thanks!

**Pull request merged**